internal/replacements/replacements_test.go

```diff
@@ -290,12 +290,69 @@ var _ = Describe("replace_misconfigured_pods", func() {
 				pod.Spec.NodeSelector = map[string]string{
 					"dummy": "test",
 				}
-				needsRemoval, err := processGroupNeedsRemovalForPod(cluster, pod, processGroup, log)
+				needsRemoval, err := processGroupNeedsRemovalForPod(cluster, pod, processGroup, log, true)
 				Expect(needsRemoval).To(BeFalse())
 				Expect(err).NotTo(HaveOccurred())
 			})
 		})
 
+		Context("when the securityContext doesn't match", func() {
+			BeforeEach(func() {
+				pClass = fdbv1beta2.ProcessClassStorage
+				remove = false
+			})
+
+			It("should need a removal for RunAsUser change", func() {
+				pod.ObjectMeta.Annotations[fdbv1beta2.LastSpecKey] = "banana"
+				// if ReplaceInstancesWhenResourcesChange is true, any spec change should result in replacement
+				cluster.Spec.ReplaceInstancesWhenResourcesChange = new(bool)
```

Comment:
This shouldn't be the case and would mean something is wrong with the test setup.